app/src/main/java/com/stevesoltys/seedvault/restore/RestoreErrorBroadcastReceiver.kt

```diff
@@ -0,0 +1,34 @@
+package com.stevesoltys.seedvault.restore
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.Intent.FLAG_ACTIVITY_NEW_TASK
+import androidx.core.net.toUri
+import com.stevesoltys.seedvault.BackupNotificationManager
+import org.koin.core.context.GlobalContext.get
+
+internal const val ACTION_RESTORE_ERROR_UNINSTALL = "com.stevesoltys.seedvault.action.UNINSTALL"
+internal const val EXTRA_PACKAGE_NAME = "com.stevesoltys.seedvault.extra.PACKAGE_NAME"
+internal const val REQUEST_CODE_UNINSTALL = 4576841
+
+class RestoreErrorBroadcastReceiver : BroadcastReceiver() {
+
+    // using KoinComponent would crash robolectric tests :(
+    private val notificationManager: BackupNotificationManager by lazy { get().koin.get<BackupNotificationManager>() }
+
+    override fun onReceive(context: Context, intent: Intent) {
+        if (intent.action != ACTION_RESTORE_ERROR_UNINSTALL) return
+
+        notificationManager.onRestoreErrorSeen()
+
+        val packageName = intent.getStringExtra(EXTRA_PACKAGE_NAME)!!
+        @Suppress("DEPRECATION")  // the alternative doesn't work for us
```

Comment:
Just curious, what is deprecated and what's the alternative?

Reply:
It is described [in the docs](https://developer.android.com/reference/android/content/Intent#ACTION_UNINSTALL_PACKAGE).

[The alternative](https://developer.android.com/reference/android/content/pm/PackageInstaller.html#uninstall(java.lang.String,%20android.content.IntentSender)) works only for:
*    the current "installer of record" for the package
*    the device owner
*    the affiliated profile owner 

and we are neither.